calico_cni/policy_drivers.py

```diff
@@ -131,13 +131,55 @@ def generate_tags(self):
         """
         return []
 
+class MesosPolicyDriver(DefaultPolicyDriver):
+    """
+    Implements default network policy for a Mesos container manager.
+    """
+    def __init__(self, network_name, network_args):
+        """
+        Extract any labels that have been specified in the mesos-namespaced labels field
+        of the network_args. Mesos CNI inserts the following additional data into
+        the args field:
+        "args" : {
+          "org.apache.mesos" : {
+            "network_info" : {
+              "name" : "mynet",
+              "labels" : {
+                "labels" : [
+                  { "key" : "app", "value" : "myapp" },
+                  { "key" : "env", "value" : "prod" }
+                ]
+              }
+            }
+          }
+        }
+        Note that Mesos labels are a list of {"key": <key>, "value": <value>}, so pull
+        the key and values and populate the labels dictionary.
+        """
+        self.labels = {}
+
+        labels_list = network_args[MESOS_NS_KEY]. \
+            get(MESOS_NETWORK_INFO_KEY, {}). \
+            get(MESOS_LABELS_OUTER_KEY, {}). \
+            get(MESOS_LABELS_KEY, [])
+
+        for label in labels_list:
```

Comment:
There's no way that Mesos might pass us multiple keys with the same value, right?

I'd assume not, but if it were theoretically possible then we'd at least want a warning in this code.  I say leave it as is, but maybe have a quick think to see if it might be possible.